src/screens/MainApp/DiagnoseRequest/screens/Gallery/index.js

```diff
@@ -0,0 +1,197 @@
+        return
+      }
+
+      if (hasMoreImages) {
+        try {
+          updateImagesAndCursor(await getPhotos(endCursor))
+        } catch (error) {
+          alertPermissionsError()
+        }
+      }
+    },
+    [endCursor, hasMoreImages]
+  )
+
+  const canSelectMoreImages = useCallback(
+    () => selectedImages.length < MAX_SELECTION_COUNT,
+    [selectedImages]
+  )
+
+  const toggleSelect = useCallback(
+    (item) => {
+      const newSelectedImages = [...selectedImages]
+      const index = newSelectedImages.indexOf(item)
+
+      if (index !== -1) {
+        newSelectedImages.splice(index, 1)
+      } else {
+        if (canSelectMoreImages()) {
+          newSelectedImages.push(item)
+        }
+      }
+
+      setSelectedImages(newSelectedImages)
+    },
+    [selectedImages]
+  )
+
+  const callCamera = useCallback(
+    async () => {
+      const granted = await requestWritePermissions()
+
+      if (granted !== PermissionsAndroid.RESULTS.GRANTED) {
+        alertPermissionsError()
+        return
+      }
+
+      try {
+        const [image] = await ImageService.openCamera()
+        const newImages = [...images]
+
+        newImages.splice(1, 0, image)
+        await CameraRoll.saveToCameraRoll(image)
+        setImages(newImages)
+      } catch (error) {
+      }
+    },
+    [images]
+  )
+
+  useEffect(() => {
+    const params = navigation.state.params
+
+    if (!params) return
+
+    if (params.addedImages) {
+      const newImages = [...images]
+      newImages.splice(1, 0, ...params.addedImages)
+      setImages(newImages)
+    }
+    if (params.selectedImages) {
+      setSelectedImages([...params.selectedImages])
+    }
+  }, [navigation.state])
+
+  const size = useMemo(
+    () => SCREEN_WIDTH / numColumns,
+    [numColumns]
+  )
+
+  return (
+    <Background>
+      <FlatList
+        data={images}
+        keyExtractor={item => item}
+        numColumns={numColumns}
+        contentContainerStyle={styles.listContainer}
+        onEndReached={loadMore}
+        renderItem={({ item, index }) => {
+          if (index === 0) {
+            return (
+              <Button onPress={callCamera} style={[styles.box, { width: size, height: size }]}>
+                <View style={styles.innerBox}>
+                  <SvgXml xml={CameraLogo} />
+                  <Body primary>Sacar foto</Body>
+                </View>
+              </Button>
+            )
+          }
+          return (
+            <ListItem
+              item={item}
+              width={size}
+              height={size}
+              selectedIndex={selectedImages.indexOf(item) + 1}
+              onPress={() => toggleSelect(item)}
+            />
+          )
+        }}
+      />
+      {(selectedImages.length >= MIN_SELECTION_COUNT) && (
+        <PrimaryButton
+          style={styles.button}
+          onPress={goToImageReview}
+        >
+          <Description white>Continuar</Description>
+        </PrimaryButton>
+      )}
```

Comment:
Likewise you can move this to a separate `goToReviewButton`.

This case is more nitpicky though, it reads fine and the possibility of it changing is low.